proto/pocket/service/relay.proto

```diff
@@ -0,0 +1,87 @@
+syntax = "proto3";
+package pocket.service;
+
+option go_package = "pocket/x/service/types";
+
+
+import "cosmos_proto/cosmos.proto";
+import "pocket/service/service.proto";
+import "pocket/application/application.proto";
+import "pocket/supplier/supplier.proto";
+
+// Relay is a protobuf containing both the RelayRequest, signed by the Application
+// and the RelayResponse, signed by the Supplier. Together, the serialized tuple is
+// what is inserted into the SMST leaves as values in the Claim/Proof lifecycle.
+message Relay {
+    RelayRequest req = 1;
+    RelayResponse res = 2;
+}
+
+// RelayMeta is used to hold the metadata for a RelayRequest.
+message RelayRequestMetadata {
+    session.SessionHeader session_header = 1; // The session header for the relay
+    map<string, string> headers = 2; // The headers of the request
+    bytes application_signature = 3; // The signature of the application of the request
+}
+
+// RelayResponse is used to hold the response from a RelayRequest.
+message RelayRequest {
+    RelayRequestMetadata meta = 1;
+    // TODO: Add other relay types in the future such as the ones below:
+    oneof payload {
+        JSONRPCPayload json_rpc_payload = 2;
+        RESTPayload rest_payload = 3;
+        // WebSocketsPayload websockets_payload = 4;
+        // GRPCPayload grpc_payload = 5;
+        // GraphQLPayload graphql_payload = 6;
+    }
+}
+
+// RelayResponseMetadata is used to hold the metadata for a RelayResponse
+message RelayResponseMetadata {
+    session.SessionHeader session_header = 1; // The session header for the relay
+    map<string, string> headers = 2; // The headers of the response
+    uint32 status_code = 3; // The status code of the response
+    string err = 4; // The error message of the response
+    bytes supplier_signature = 5; // The signature of the supplier on the response
+}
+
+// RelayResponse is used to hold the response to a RelayRequest.
+message RelayResponse {
+    RelayResponseMetadata meta = 1;
+    // TODO: Add other relay types in the future such as the ones below:
+    oneof payload {
```

Comment:
`RelayResponse` should have `*Payload` on their own. eg. a `JSONRPCPayload(Response)` does not contain `method` or `params` but has `error` and `result`.

In the REST case, a response does not return the `path` (http_path) or `method` (RESTRequestType)